third_party/terraform/utils/config_test.go

```diff
@@ -90,6 +90,76 @@ func TestAccConfigLoadValidate_credentials(t *testing.T) {
 	}
 }
 
+func TestAccConfigLoadValidate_impersonated(t *testing.T) {
+	if os.Getenv(TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Network access not allowed; use %s=1 to enable", TestEnvVar))
+	}
+	testAccPreCheck(t)
+
+	serviceaccount := multiEnvSearch([]string{"IMPERSONATE_SERVICE_ACCOUNT_ACCTEST"})
+	creds := getTestCredsFromEnv()
+	proj := getTestProjectFromEnv()
+
+	config := &Config{
+		Credentials:               creds,
+		ImpersonateServiceAccount: serviceaccount,
+		Project:                   proj,
+		Region:                    "us-central1",
+	}
+
+	ConfigureBasePaths(config)
+
+	err := config.LoadAndValidate(context.Background())
+	if err != nil {
+		t.Fatalf("error: %v", err)
+	}
+
+	_, err = config.NewComputeClient(config.userAgent).Zones.Get(proj, "us-central1-a").Do()
+	if err != nil {
+		t.Fatalf("expected API call with loaded config to work, got error: %s", err)
+	}
+}
+
+func TestAccConfigLoadValidate_accessTokenImpersonated(t *testing.T) {
+	if os.Getenv(TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Network access not allowed; use %s=1 to enable", TestEnvVar))
+	}
+	testAccPreCheck(t)
+
+	creds := getTestCredsFromEnv()
+	proj := getTestProjectFromEnv()
+	serviceaccount := multiEnvSearch([]string{"IMPERSONATE_SERVICE_ACCOUNT_ACCTEST"})
+
+	c, err := google.CredentialsFromJSON(context.Background(), []byte(creds), testOauthScope)
```

Comment:
One last thing @rileykarson found, we need to pass the `DefaultClientScopes` along with `testOauthScope` here, I think thats why this test was failing.

Reply:
Hmm, i'll replace testOauthScope with DefaultClientScopes. Instead of appending the the two, you can use the default one as it in includes compute + cloud-platform